### DISCUSSION TITLE
Activate user accounts before first use

### DISCUSSION THREAD

**Discussion by schneidr** [ltb-project/self-service-password#961] [Q&A]
We would like to create LDAP accounts that are disabled at first and only get activated after the user reads the TOS and sets a password. Could self-service-password somehow be utilized to implement this?

Reset by [mail token](https://self-service-password.readthedocs.io/en/latest/config_tokens.html) wouldn't work because it relies on the users PHP session.

The [REST API options](https://self-service-password.readthedocs.io/en/latest/webservices.html) don't seem to be so helpful either.

Are there any other options?

**Comment by coudot** [comment]
Yes you can use the REST API to force a new password to user: https://self-service-password.readthedocs.io/en/latest/webservices.html#post--rest-v1-adminchangepassword.php

Of course you need to implement your UI for the TOS and to get password of a user.

If the question was to have this UI inside SSP, it indeed does not exist.
